Internal Review (#4)
jeel38/owasp-mastg: demos/android/MASVS-CODE/MASTG-DEMO-0054/MASTG-DEMO-0054.md

> id: MASTG-DEMO-0054

Reviewer: The Demo number already exist ,change to lastest.

code: [kotlin]

> test: MASTG-TEST-0281

Reviewer: Change this as well to the lastest, and also change the test file name accordingly.

profiles: [L1, L2]
---

### Sample

> The code snippet demonstrates an insecure use of `SharedPreferences` where data is loaded without a proper integrity check, which is a form of input validation for stored data.

Reviewer: ```suggestion
The code snippet demonstrates the insecure use of SharedPreferences, as data is loaded without an integrity check.
```

{{ MastgTest.kt # MastgTest_reversed.java }}

### Steps

Let's run @MASTG-TOOL-0110 rules against the sample code.

{{ ../../../../rules/mastg-android-local-storage-input-validation.yml }}

{{ run.sh }}

### Observation

> The output file correctly identifies the vulnerable pattern where data is loaded without being validated.

Reviewer: ```suggestion
The rule identifies that data is being loaded without being validated.
```

{{ output.txt }}

### Evaluation

> The test fails because the rule detected that the app does NOT use an `HMAC` integrity check together with `SharedPreferences`.

Reviewer: ```suggestion
The test fails as the code does not use an `HMAC` integrity check together with `SharedPreferences` data.
```
